lib/upipe-x265/upipe_x265.c

```diff
@@ -230,6 +243,132 @@ UPIPE_HELPER_UBUF_MGR(upipe_x265, ubuf_mgr, flow_format, ubuf_mgr_request,
                       upipe_x265_unregister_output_request)
 UPIPE_HELPER_UCLOCK(upipe_x265, uclock, uclock_request, NULL, upipe_throw_provide_request, NULL)
 
+static int upipe_x265_param_parse(struct upipe *upipe,
+                                  x265_param *params,
+                                  const char *name,
+                                  const char *value)
+{
+    struct upipe_x265 *upipe_x265 = upipe_x265_from_upipe(upipe);
+
+    if (unlikely(upipe_x265->api == NULL || name == NULL))
+        return UBASE_ERR_INVALID;
+
+    int ret = upipe_x265->api->param_parse(params, name, value);
+    if (unlikely(ret < 0)) {
+        const char *reason = "";
+        if (ret == X265_PARAM_BAD_NAME)
+            reason = ": bad name";
+        else if (ret == X265_PARAM_BAD_VALUE)
+            reason = ": bad value";
+        upipe_err_va(upipe, "cannot set option %s=%s%s",
+                     name, value ?: "true", reason);
+        return UBASE_ERR_EXTERNAL;
+    }
+
+    return UBASE_ERR_NONE;
+}
+
+/** @internal @This sets the content of an x265 option.
+ * upipe_x265_reconfigure must be called to apply changes.
+ *
+ * @param upipe description structure of the pipe
+ * @param option name of the option
+ * @param content content of the option
+ * @return an error code
+ */
+static int upipe_x265_set_option(struct upipe *upipe,
+                                 const char *name,
+                                 const char *value)
+{
+    struct upipe_x265 *upipe_x265 = upipe_x265_from_upipe(upipe);
+    return upipe_x265_param_parse(upipe, &upipe_x265->params, name, value);
+}
+
+static void apply_params(struct upipe *upipe, x265_param *params)
+{
+    struct upipe_x265 *upipe_x265 = upipe_x265_from_upipe(upipe);
+    struct uref *flow_def = upipe_x265->flow_def_input;
+    const char *value;
+
+    if (flow_def == NULL)
+        return;
+
+    params->logLevel = X265_LOG_DEBUG;
+
+    struct urational fps = {0, 0};
+    if (likely(ubase_check(uref_pic_flow_get_fps(flow_def, &fps)))) {
+        params->fpsNum = fps.num;
+        params->fpsDenom = fps.den;
+    }
+
+    params->vui.aspectRatioIdc = upipe_x265->aspect_ratio_idc;
+    if (params->vui.aspectRatioIdc == X265_EXTENDED_SAR) {
+        params->vui.sarWidth = upipe_x265->sar_width;
+        params->vui.sarHeight = upipe_x265->sar_height;
+    }
+    params->vui.bEnableOverscanInfoPresentFlag = upipe_x265->overscan;
+    params->vui.bEnableOverscanAppropriateFlag = upipe_x265->overscan;
```

Comment:
Actually, the correct behaviour should be:
 - if the attribute is not present, InfoPresentFlag = false
 - if the attribute is present, InfoPresentFlag = true; AppropriateFlag = overscan